### PR TITLE
74322 - download enc data from fss for aio exchange set

### PR DESCRIPTION
**Pull request by shailen14599**
- Done code changes (Restructuring of existing code for exchange set and large media exchange set and added additional code for aio ES)
- Added unit test cases
- Added functional tests

### CODE REVIEW

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetWhenAioIsEnabled.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        {
+            foreach (string batchId in Config.AIOConfig.AioExchangeSetBatchIds)
+            {
+                DownloadedFolderPath = await FileContentHelper.DownloadAndExtractAioZip(FssJwtToken, batchId);
+
+                int count = Directory.GetFiles(Path.Combine(DownloadedFolderPath, Config.AIOConfig.AioEncTempPath)).Length;
+                Assert.IsTrue(count > 0, $"Downloaded Enc files count is 0, Instead of expected count should be greater than 0");
+
+                FileContentHelper.DeleteDirectory(Config.AIOConfig.AioExchangeSetFileName);
+            }
```
Comment (ray-sj) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetWhenAioIsEnabled.cs:36]: are we not missing additional tests to cover failed download

**Review by ray-sj — CHANGES_REQUESTED** [review]

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetWhenAioIsEnabled.cs (hunk 1 above)
Comment (Mayuresh-Gawde) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetWhenAioIsEnabled.cs:36]: Hi Ray,

The actual download functionality is part of the future PBIs, and the scenario will get covered under those. As part of this PBI we have used the static batches. I hope that's fine.

Regards,
Mayuresh Gawde

**Review by ray-sj — APPROVED** [review]

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs [hunk 2]
```diff
@@ -280,7 +285,23 @@ private async Task CheckProductOrCancellationData(SearchBatchResponse internalSe
...
+                        {
+                            await PerformBatchFileDownload(item, productItem, exchangeSetRootPath, message, cancellationTokenSource, cancellationToken);
+                        }
+                    }
+                    else
+                    {
+                        await PerformBatchFileDownloadForLargeMediaExchangeSet(item, productItem, exchangeSetRootPath, message, cancellationTokenSource, cancellationToken);
+                    }
+                }
                 else
```
Comment (DanOwenUKHO) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs:305]: Not your change but can you add brackets around this for the scope of this else

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs [hunk 3]
```diff
@@ -280,7 +285,23 @@ private async Task CheckProductOrCancellationData(SearchBatchResponse internalSe
                 internalSearchBatchResponse.Entries.Add(item);
                 productList.Add(compareProducts);
                 if (CommonHelper.IsPeriodicOutputService)
```
Comment (DanOwenUKHO) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs:287]: This logic from 288-307 should be split out into its own method to reduce the complexity in  `CheckProductOrCancellationData`

**Review by DanOwenUKHO — CHANGES_REQUESTED** [review]
A couple of comments to review but looks good otherwise.
Stryker report: 44.48% -> 43.99%, can we please raise this to atleast 44.48%

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs (hunk 2 above)
Comment (shailen14599) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs:307]: Resolved

**Review by DanOwenUKHO — APPROVED** [review]

**Pull request merged**